### PR TITLE
[Serve] support -h to serve cli

### LINKED ISSUE #35734: [serve] support `-h` in addition to `--help`
`serve -h` fails, this is a common pattern.

Relevant docs:
https://click.palletsprojects.com/en/8.1.x/documentation/#help-parameter-customization

### PR DESCRIPTION
**Pull request by GeneDer**
## Why are these changes needed?

- `-h` is a common pattern and is not currently supported on `serve` cli. This PR adds the support to call `serve -h`

## Related issue number

Closes #35734

## Checks

- [x] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(
   - [x] Tested by running `python python/ray/setup-dev.py -y` then seeing `serve -h` working

### CODE REVIEW

File: python/ray/setup-dev.py [hunk 1]
```diff
@@ -122,6 +122,7 @@ def do_link(package, force=False, skip_list=None, local_path=None):
         skip_list=args.skip,
         local_path="../../../dashboard",
     )
+    do_link("serve", force=args.yes, skip_list=args.skip)
```
Comment (edoakes) [inline python/ray/setup-dev.py:125]: Could you please do this in a separate PR? Better to keep things orthogonal where possible, makes it easier to track down issues if they happen.

In this case it's pretty minor, but better to be disciplined about it.

**Review by edoakes — APPROVED** [review]

File: python/ray/serve/scripts.py [hunk 2]
```diff
@@ -122,7 +123,10 @@ def convert_args_to_dict(args: Tuple[str]) -> Dict[str, str]:
     return args_dict
 
 
-@click.group(help="CLI for managing Serve applications on a Ray cluster.")
+@click.group(
+    help="CLI for managing Serve applications on a Ray cluster.",
+    context_settings=CONTEXT_SETTINGS,
```
Comment (shrekris-anyscale) [inline python/ray/serve/scripts.py:128]: [Nit] Could you pass in the `CONTEXT_SETTING` dictionary directly instead of using a constant? Afaik, this dictionary is only used in this argument, so I don't think it needs to be stored as a global constant.
